crypto/ocsp/ocsp_test.cc

```diff
@@ -197,116 +193,97 @@ static void CheckOCSP_CERTSTATUS(
   int expected_ocsp_verify_status;
   int expected_ocsp_resp_find_status;
   int expected_ocsp_cert_status;
   const char *expected_ocsp_cert_status_string;
 };
 
-static const OCSPTestVectorExtended nTestVectors[] = {
+static const OCSPAWSTestVector nTestVectors[] = {
     // === SHA1 OCSP RESPONSES ===
     // Test valid OCSP response signed by an OCSP responder.
     {"ocsp_response", "ca_cert", "server_cert", EVP_sha1(),
-     OCSP_RESPONSE_STATUS_SUCCESSFUL, "successful", OCSP_VERIFYSTATUS_SUCCESS,
-     OCSP_RESPFINDSTATUS_SUCCESS, V_OCSP_CERTSTATUS_GOOD, "good"},
+     OCSP_VERIFYSTATUS_SUCCESS, OCSP_RESPFINDSTATUS_SUCCESS,
+     V_OCSP_CERTSTATUS_GOOD, "good"},
     // Test against same good OCSP response, but checking behavior of not
     // specifying hash algorithm used for |OCSP_cert_to_id| this time (should
     // default to sha1). When |*dgst| is set to NULL, the default hash algorithm
     // should automatically be set to sha1. The revocation status check of the
     // response should work if hash algorithm of |cert_id| has been set to sha1
     // successfully.
     {"ocsp_response", "ca_cert", "server_cert", nullptr,
-     OCSP_RESPONSE_STATUS_SUCCESSFUL, "successful", OCSP_VERIFYSTATUS_SUCCESS,
-     OCSP_RESPFINDSTATUS_SUCCESS, V_OCSP_CERTSTATUS_GOOD, "good"},
+     OCSP_VERIFYSTATUS_SUCCESS, OCSP_RESPFINDSTATUS_SUCCESS,
+     V_OCSP_CERTSTATUS_GOOD, "good"},
     // Test valid OCSP response directly signed by the CA certificate.
     {"ocsp_response_ca_signed", "ca_cert", "server_cert", EVP_sha1(),
-     OCSP_RESPONSE_STATUS_SUCCESSFUL, "successful", OCSP_VERIFYSTATUS_SUCCESS,
-     OCSP_RESPFINDSTATUS_SUCCESS, V_OCSP_CERTSTATUS_GOOD, "good"},
+     OCSP_VERIFYSTATUS_SUCCESS, OCSP_RESPFINDSTATUS_SUCCESS,
+     V_OCSP_CERTSTATUS_GOOD, "good"},
     // Test OCSP response status is revoked.
     {"ocsp_response_revoked", "ca_cert", "server_cert", EVP_sha1(),
-     OCSP_RESPONSE_STATUS_SUCCESSFUL, "successful", OCSP_VERIFYSTATUS_SUCCESS,
-     OCSP_RESPFINDSTATUS_SUCCESS, V_OCSP_CERTSTATUS_REVOKED, "revoked"},
+     OCSP_VERIFYSTATUS_SUCCESS, OCSP_RESPFINDSTATUS_SUCCESS,
+     V_OCSP_CERTSTATUS_REVOKED, "revoked"},
     // Test OCSP response status is unknown.
     {"ocsp_response_unknown", "ca_cert", "server_cert", EVP_sha1(),
-     OCSP_RESPONSE_STATUS_SUCCESSFUL, "successful", OCSP_VERIFYSTATUS_SUCCESS,
-     OCSP_RESPFINDSTATUS_SUCCESS, V_OCSP_CERTSTATUS_UNKNOWN, "unknown"},
+     OCSP_VERIFYSTATUS_SUCCESS, OCSP_RESPFINDSTATUS_SUCCESS,
+     V_OCSP_CERTSTATUS_UNKNOWN, "unknown"},
     // for the requested certificate. (So this would be a completely valid
     // response to a different OCSP request for the other certificate.)
     {"ocsp_response", "ca_cert", "server_ecdsa_cert", EVP_sha1(),
-     OCSP_RESPONSE_STATUS_SUCCESSFUL, "successful", OCSP_VERIFYSTATUS_SUCCESS,
-     OCSP_RESPFINDSTATUS_ERROR, 0, nullptr},
+     OCSP_VERIFYSTATUS_SUCCESS, OCSP_RESPFINDSTATUS_ERROR, 0, nullptr},
```

Comment:
Just curious why do some of these use a define like V_OCSP_CERTSTATUS_UNKNOWN, and here you use a 0? What happens if you get the status for something that fails? I'm also a little concerned since 0 is V_OCSP_CERTSTATUS_GOOD if I'm looking in the right spot.

Reply:
Yeah that's on me for using `0` which has the same value as `V_OCSP_CERTSTATUS_GOOD`. I should of used `-1` so it doesn't coincide with any of the other CERTSTATUS codes. I'll change it.
* OCSP_CERTSTATUS has three different possibilities: "good", "revoked", and "unknown". This value is extracted with `OCSP_resp_find_status`.
* `OCSP_resp_find_status` has to succeed in order to extract a status, so it's impossible to get a "cert status" for something that fails. It'll result in an error if it's ill-defined, which is why I was defining it with `0` since there was essentially no value to compare to.